Improve relevance scoring for titles and object-name matches in search results (#12441)
sphinx-doc/sphinx: CHANGES.rst

* #11961: Omit anchor references from document title entries in the search index,
  removing duplication of search results.
  Patch by James Addison.

> * #12391: Adjust scoring of matches during HTML search so that document main
>   titles tend to rank more highly than subsection titles, and also to provide
>   a gain to matches on the name of programming domain objects relative to
>   title/subtitle matches.
>   Patch by James Addison and Will Lachance.

Reviewer: ```suggestion
* #12391: Adjust scoring of matches during HTML search so that main
  titles tend to rank higher than subsection titles. In addition, 
  boost matches on the name of programming domain objects relative
  to title/subtitle matches.
  Patch by James Addison and Will Lachance.
```
